sympy/integrals/tests/test_integrals.py

```diff
@@ -1925,29 +1929,42 @@ def test_pr_23583():
     assert integrate(1/sqrt((x - I)**2-1)) == Piecewise((acosh(x - I), Abs((x - I)**2) > 1), (-I*asin(x - I), True))
 
 
+def test_issue_22863():
+    i = integrate((3*x**3-x**2+2*x-4)/sqrt(x**2-3*x+2), (x, 0, 1))
+    assert i == -101*sqrt(2)/8 - 135*log(3 - 2*sqrt(2))/16
+    assert math.isclose(i.n(), -2.98126694400554)
+
+
 def test_hyperbolic():
     assert integrate(coth(x)) == x - log(tanh(x) + 1) + log(tanh(x))
     assert integrate(sech(x)) == 2*atan(tanh(x/2))
     assert integrate(csch(x)) == log(tanh(x/2))
 
 
 def test_sqrt_quadratic():
+    nc = Symbol('c', nonzero=True)
     assert integrate(1/sqrt(3*x**2+4*x+5)) == sqrt(3)*asinh(3*sqrt(11)*(x + S(2)/3)/11)/3
     assert integrate(1/sqrt(-3*x**2+4*x+5)) == sqrt(3)*asin(3*sqrt(19)*(x - S(2)/3)/19)/3
     assert integrate(1/sqrt(3*x**2+4*x-5)) == sqrt(3)*log(6*x + 2*sqrt(3)*sqrt(3*x**2 + 4*x - 5) + 4)/3
-    assert integrate(1/sqrt(a+b*x+c*x**2), x) == log(2*sqrt(c)*sqrt(a+b*x+c*x**2)+b+2*c*x)/sqrt(c)
+    i = integrate(1/sqrt(a+b*x+c*x**2), x)
+    assert i.subs(c, nc).subs(nc, c) == log(2*sqrt(c)*sqrt(a+b*x+c*x**2)+b+2*c*x)/sqrt(c)
```

Comment:
I think it's better to test for exactly what this returns if it does return what we want it to. Otherwise if we only want to know what happens when `nonzero=True` is set then why not just use `nc` directly in the test?

Reply:
Also note that `nonzero=True` implies real so if you intended for this to be an arbitrary nonzero complex number then it should be declared as `Symbol('c', complex=True, zero=False)`.

Reply:
I didn't want to use a big `Piecewise` because it's ugly. Separating the result to 2 assertions makes it a bit clearer and when code changes we know easily which branch fails.
I didn't realize the meaning of `nonzero`. I'll change that.